cloud/amazon/emr.py

```diff
@@ -0,0 +1,420 @@
+            action_on_failure: "TERMINATE_JOB_FLOW"
+
+          - name: Setup Mysql JAR
+            jar_uri: "s3://{{ region }}.elasticmapreduce/libs/script-runner/script-runner.jar"
+            args: "{{ s3_distro_path }}/scripts/setup_mysql_jar.sh,{{ mysql_connector_jar_version }}"
+            action_on_failure: "TERMINATE_JOB_FLOW"
+
+          - name: Run Custom JAR
+            jar_uri: "s3://bucket/path/to/custom.jar"
+            main: "MainClass"
+            args: "arg1,arg2"
+            action_on_failure: "CONTINUE"
+
+        # Add as many bootstrap actions as needed here.
+        # args are separated by commas.
+        bootstrap_actions:
+          - name: set_memory_intensive
+            action: '{{ bootstrap_actions_path }}/configurations/latest/memory-intensive'
+
+          - name: add_swap
+            action: '{{ bootstrap_actions_path }}/add-swap'
+            args: '2048'
+
+          - name: configure_hadoop
+            action: '{{ bootstrap_actions_path }}/configure-hadoop'
+            args: '-c,hadoop.proxyuser.oozie.hosts=*,-c,hadoop.proxyuser.oozie.groups=*,-c,hadoop.proxyuser.hadoop.hosts=*,-c,hadoop.proxyuser.hadoop.groups=*,-h,dfs.permissions=false,-c,fs.s3n.multipart.uploads.enabled=true,-c,fs.s3n.multipart.uploads.split.size=134217728,-s,mapred.tasktracker.map.tasks.maximum=8,-s,mapred.tasktracker.reduce.tasks.maximum=2,-s,mapred.min.split.size=134217728,-s,mapred.max.split.size=536870912,-m,mapred.job.reuse.jvm.num.tasks=-1'
+
+          - name: sleep_if_master
+            action: '{{ bootstrap_actions_path }}/run-if'
+            args: 'instance.isMaster=true,sleep 60'
+
+          - name: optimization_setup
+            action: '{{ bootstrap_actions_path }}/run-if'
+            args: 'instance.isMaster=true,{{ bootstrap_scripts_dir }}/setup_optimization.sh,{{ ls_config_tar_filename }},{{ ls_dist_tar }},{{ s3_distro_root }},{{ cluster_name }},{{ region }}'
+
+          - name: configure_module_props
+            action: '{{ bootstrap_actions_path }}/run-if'
+            args: 'instance.isMaster=true,{{ bootstrap_scripts_dir }}/config_module_props.sh'
+
+          - name: setup_oozie
+            action: '{{ bootstrap_actions_path }}/run-if'
+            args: 'instance.isMaster=true,{{ bootstrap_scripts_dir }}/setup_oozie.sh,{{ oozie_version }}'
+
+          - name: setup_classpath
+            action: '{{ bootstrap_scripts_dir }}/setup_classpath.sh'
+
+          - name: setup_hadoop_heap
+            action: '{{ bootstrap_scripts_dir }}/setup_hadoop_heap.sh'
+'''
+
+import sys
+import time
+from datetime import datetime
+
+try:
+    import boto.emr
+    from boto.emr.connection import EmrConnection
+    from boto.emr.instance_group import InstanceGroup
+    from boto.emr.step import JarStep
+    from boto.emr.step import InstallHiveStep
+    from boto.emr.bootstrap_action import BootstrapAction
+except ImportError:
```

Comment:
modules should not exit, set a variable and use the json_fail, look at other aws modules for an example (HAS_BOTO)